The study of organisms in Biology is a highly complex complex process
involving many disciplines. To better understand how these organisms
function, we must braker down the problem into different
sub-problems. One important sub-problem in biology is the
understanding of the molecular tools and processes used by cells to
function in normal and abnormal environmental scenarios or stress
conditions. These conditions may include disease and environmental
stress for example. However, previous work in biology has shown that
the underlying backbone of information, known as the genome, can vary
widely between different organisms in many aspects, such as overall
structure, length, ploidy, methylation, and overall gene content. All
of these aspects can affect the survival of an organism in any given
environment, some of which may be interesting to researchers. As an
example, one of the most popular and extensively studied diseases is
cancer. Through the study of human genomes, researchers identified a
key gene, named TP53, involved in the suppression of
tumours. Functional mutations affecting this gene can result in
increased risk of cancer. Understanding how and why TP53 suppresses
tumours can provide insight into future cancer prevention methods and
treatments. These genes can then be mapped to the genome of the
organism to identify its location.

This general workflow can be applied to features of interest from
organisms in all branches of life. To facilitate this process of
identifying genes from a genome, the process of genome annotation was
developed. In this case, instead of identifying a gene of interest and
then mapping it back to the genome, gene finders identify potential
genes from a reference genome before truly knowing their function or
if the candidate is truly untilized by the organisms. This set of
potential genes acts as a reference for future research. However, to
generate a reliable set of possibe genes, a gene finder must be
supplied with a suitable high quality genome. In many cases,
researchers may be studying a specific strain or variety of organism
that differs from the reference assembly and annotation available for
the organism of interest. Rather than using the reference assembly for
analysis, it may be beneficial to generate a new assembly for the
unique variety or strain, which must then be passed through a gene
finding tool. With the variety of gene finding tools and approaches,
the choice of an appropriate tool can affect the resulting gene
set. This problem raises a question. Do the results from gene finding
tools differ? And more importantly, how should one compare results
from gene finding tools when there is no reference annotation for a
specifc variety in question? Most genome annotation tools benchmark
their performance in comparison to an existing reference annotation,
which is usually considered to be of suitably high quality. This is
not possible in the case of unique assemblies, and so the devleopment
of a comparative methodology is in order.
