\textit{Trichoderma} species are fungi from the Hypocreaceae family
 that are commonly found in soil and aid in the decomposition of plant matter.
 They are also known for their ability to produce a variety of enzymes involved in a number of different roles, including cellulose degradation, nitrogen fixation, antibody production, and cellular signalling. Some of these species are also used in agriculture as biocontrol agents against plant pathogens. Some the enzymes responsible for these roles are produced in large quantities, making them of interest to researchers in both industry and academia. In particular, the elevated production of secondary metabolites in \textit{Trichoderma} species has been shown to have a positive effect on plant growth and health, making them an attractive option for inoculation of agricultural crops. The genomes of \textit{Trichoderma} species contain a wealth of information about the genes and proteins involved in these processes, but they must first be annotated to identify the genes and their functions, which is a non-trivial task. In addition, there are a number of different tools available for genome annotation, each with its own strengths and weaknesses, making it difficult for a researcher to choose the right tool for their needs. Gene finding tools are commonly compared to one another in common model organisms, but there is little work done comparing these tools in fungi, and even fewer in \textit{Trichoderma} species. A focus on \textit{Trichoderma} is important in this case, as the gene finders rely on gene models that may not be representative of the genes found in these species.
  The sequencing of two novel \textit{Trichoderma} genomes, DC1 and Tsth20, provides an opportunity to compare the performance of several gene finding tools in these species. Results from this work may be of interest to researchers studying crop resistance and plant growth promotion, as well as those interested in the mechanisms at play behind other benefits of \textit{Trichoderma} species. In addition, this work will provide insight into the performance of gene finding tools in fungi and the evolutionary processes at play in these species.

 The main goal of this work is to compare the performance of several gene finding tools in the genomes of two novel \textit{Trichoderma} species, DC1 and Tsth20. In addition to these two genomes, we also use the RefSeq assemblies and annotations of three other \textit{Trichoderma} species, \textit{T. reesei, T. harzianum} and \textit{T. virens}, as an additional basis for comparison, with the goal of comparing these tools in a generic context, rather than in the context of a gold standard, as a gold standard is not available for these species. The exact research questions pursued in this work are explained in detail in Chapter~\ref{chap:research-questions}. 

The general structure of this thesis is as follows. In Chapter~\ref{chap:background}, we provide background information on \textit{Trichoderma} species, genome assembly and annotation, and gene finding tools. As mentioned before, the research questions are explored in Chapter~\ref{chap:research-questions}, while in Chapter~\ref{chap:methods}, we describe the methods used to answer those questions. In Chapter~\ref{chap:results}, we present the results of our analyses, including comparisons of the assemblies and annotations of the DC1 and Tsth20 genomes to those of \textit{T. reesei, T. harzianum} and \textit{T. virens}, as well as comparisons of the performance of the gene finding tools. Finally, in Chapter~\ref{chap:conclusions}, we discuss the implications of our findings, as well as potential future work in this area. We make note that in this thesis, we refer to annotations from NCBI's RefSeq database as `RefSeq', although the annotations may have been generated using a variety of tools and methods. 
