\section{Genes in Regions of Anomalous GC Content}\label{section:gc-regions}

To better understand the characteristics of AT-rich sequences in these assemblies, the same sliding window approach described in Section~\ref{section:assemblies} was used to identify segments of genomic sequence with greater than 72\% AT nucleotide content (less than 28\% GC content). Table~\ref{table:gc-content} shows the total length of AT-rich sequence identified in each assembly along with the number of segments identified and the proportion of the assembly comprised of AT-rich sequence. From these results, we see that DC1 and \textit{T. harzianum} have the largest proportion of AT-rich sequence, while Tsth20 and \textit{T. reesei} have moderate amounts of AT-rich sequence. \textit{T. virens} has very little AT-rich sequence compared to the other assemblies.

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|c|}
      \hline
      Genome & Total Length (bp) & Low GC Segments & Low GC Length (bp) & Proportion (\%) \\ \hline
      DC1 & 38,616,239 & 610 & 2,064,202 & 5.35 \\ \hline
      Tsth20 & 41,588,851 & 245 & 891,216 & 2.14 \\ \hline
      \textit{T. reesei} & 33,395,713 & 689 & 1,269,347 & 3.80 \\ \hline
      \textit{T. harzianum} & 40,980,648 & 1,311 & 2,527,773 & 6.17 \\ \hline
      \textit{T. virens} & 39,022,666 & 96 & 162,356 & 0.42 \\ \hline
    \end{tabular}
  \end{center}
  \caption[AT-rich sequence content in Trichoderma assemblies]{AT-rich sequence content (windows with $>$ 72\% AT nucleotides) in each \textit{Trichoderma} genome assembly.}\label{table:gc-content}
\end{table}

Figure~\ref{fig:gc-regions} and Table~\ref{table:gc-regions} shows the
results of applying the same region finding process as described in Section~\ref{section:region-met} to
segments of the genome identified as AT-rich. We see that there
are few regions with gene predictions in AT-rich genomic
segments. In DC1 and Tsth20, there are very few regions with full support
from both Braker2 and GeneMark, but more singletons. Again, as in
Section~\ref{section:regions}, there are no regions with partial
support as only two gene finding tools were applied to those
assemblies. \textit{T. reesei} and \textit{T. harzianum} report more
regions in AT-rich genomic sequence than the other assemblies, but
with the majority of regions belonging to the singleton
category. \textit{T. virens} is an interesting case, reporting a
similar number of regions and genes in AT-rich genomic sequence as DC1
and Tsth20. The likely reason for this can be seen in Table~\ref{table:gc-content}, where \textit{T. virens} is shown to have very little AT-rich genomic sequence compared to the other assemblies, which may lead to fewer opportunities for singleton predictions. \textit{T. virens} is also the only assembly to report zero singleton gene predictions.

\begin{figure}
  \centering
  \includegraphics[width=0.90\textwidth]{figures/atrich-regions-barplot.pdf}
  \caption[Regions of agreement in AT-rich genomic sequence]{Regions of full, partial, and no agreement in AT-rich genomic sequence. It is important to note that in the cases of DC1 and Tsth20, full support indicates supporting gene predictions from both GeneMark and Braker2 and as such, there are no regions with partial support.}\label{fig:gc-regions}
\end{figure}

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|c|}
      \hline
      Assembly & Full Support & Partial Support & Singletons & Total Genes \\ \hline
      DC1 & 11 & N/A & 20 & 42  \\ \hline
      Tsth20 & 2 & N/A & 9 & 13  \\ \hline
      \textit{T. reesei} & 25 & 18 & 54 & 194  \\ \hline
      \textit{T. harzianum} & 26 & 43 & 68 & 265  \\ \hline
      \textit{T. virens} & 8 & 11 & 0 & 49  \\ \hline
    \end{tabular}
  \end{center}
  \caption[Agreement of gene predictions in AT-rich regions]{Regions of full and partial agreement as well as singleton regions in AT-rich genomic sequence. It is important to note that in the cases of DC1 and Tsth20, full support indicates supporting gene predictions from both GeneMark and Braker2. Column five shows the total number of genes from all gene finders in AT-rich regions for each assembly.}\label{table:gc-regions}
\end{table}

In addition to a breakdown of regions in AT-rich genomic sequence,
understanding which gene finders predict more or fewer genes in these
regions may be of interest. It is possible that an HMM, trained on
genomic sequence with varying nucleotide content, may predict genes
differently to an HMM trained only on sequences with uniformly
distributed nucleotide composition as the variation in nucleotide
composition may affect the various states and relationships between
them in an HMM. Figure~\ref{fig:gc-gene-counts} and Table~\ref{table:gc-gene-counts} shows the number of
genes predicted by each gene finding tool in regions of AT-rich
genomic sequence. GeneMark appears to predict the fewest genes in
AT-rich regions, while RefSeq appears to predict the most. Braker2
lies somewhere in the middle. Again, \textit{T. virens} appears as an
odd case, with very few predictions from all gene finders, which is likely a result of the very small amount of AT-rich genomic sequence present in that assembly.

\begin{figure}
  \centering
  \includegraphics[width=0.90\textwidth]{figures/atrich-genes-barplot.pdf}
  \caption[Gene counts in AT-rich regions]{Number of genes predicted by each gene finder in AT-rich genomic sequence.}\label{fig:gc-gene-counts}
\end{figure}

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|}
      \hline
      Assembly & Braker2 & GeneMark & RefSeq \\ \hline
      DC1 & 31 & 11 & N/A \\ \hline
      Tsth20 & 11 & 2 & N/A \\ \hline
      \textit{T. reesei} & 39 & 48 & 107 \\ \hline
      \textit{T. harzianum} & 81 & 30 & 154 \\ \hline
      \textit{T.virens} & 21 & 8 & 20 \\ \hline
    \end{tabular}
  \end{center}
  \caption[Number of genes predicted in AT-rich regions]{Number of genes predicted by Braker2, GeneMark and RefSeq
    in AT-rich genomic sequence from each assembly.}\label{table:gc-gene-counts}
\end{table}

Finally, to test the probability of any given gene prediction falling
in an AT-rich genomic sequence, a two-sided binomial test was
performed to determine if the number of genes predicted in AT-rich
sequences is proportional to the fraction of genomic sequence they
comprise. The null hypothesis in this case is that the gene finding tools predict the same proportion of genes in AT-rich genomic sequences as they do in typical genomic sequence. The results of the test are shown in Table~\ref{table:gc-binomial}. In all cases, it appears that the null hypothesis is rejected, meaning that the gene finders do not predict genes in AT-rich genomic sequence at the same rate as they do in typical genomic sequence. 

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|c|c|}
      \hline
      Tool & DC1 & Tsth20 & \textit{T. reesei} & \textit{T. harzianum} & \textit{T. virens} \\ \hline
      Braker2 & $9.56*10^{-181}$ & $1.14*10^{-259}$ & $2.68*10^{-96}$ & $4.05*10^{-140}$ & $1.35*10^{-35}$ \\ \hline
      GeneMark & $5.12*10^{-216}$ & $0.0$ & $5.66*10^{-49}$ & $5.37*10^{-219}$ & $5.31*10^{-35}$ \\ \hline
      RefSeq & N/A & N/A & $1.29*10^{-49}$ & $2.44*10^{-205}$ & $7.40*10^{-33}$ \\ \hline
    \end{tabular}
  \end{center}
  \caption[Binomial test results]{\textit{p}-values produced from a two-sided binomial test
    for each combination of tool and assembly.}\label{table:gc-binomial}
\end{table}

In summary, very few regions with gene predictions are present in
AT-rich genomic sequence. Additionally, genes predicted in these
regions tend to be isolated and not supported by other gene finders,
although some agreement is observed. In terms of number of genes
predicted, RefSeq tends to predict the most genes in these AT-rich
regions while GeneMark predicts the fewest. It was also observed that the proportion of genes predicted in AT-rich genomic sequence is not proportional to the amount of AT-rich genomic sequence present in the assemblies. In addition, the number of genes predicted in AT-rich genomic sequence varies among the different assemblies, even when the proportions of AT-rich sequence content are similar, as in the case of DC1 and \textit{T. harzianum}, although only two gene finding tools were applied to DC1. These observations suggest that gene prediction in AT-rich genomic sequence is a challenging task, and further research is required to better understand the factors influencing gene prediction in these regions.
