\documentclass[t]{beamer}
\usepackage{lmodern} % Need to use this package for proper font encoding.
\usepackage[T1]{fontenc}
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage{multicol}
\usepackage{booktabs}
\usepackage{bookmark}
\usepackage[table]{xcolor}
% Title slide information
\title{Evaluation of Gene Finding Tools When Applied to \textit{Trichoderma} Genomes - Update}
\date{July, 2025} 
\author{Connor Burbridge}
\titlegraphic{UsaskAerial.jpg} 
\institute{Department of Computer Science} 
\usetheme{usask}

\begin{document}

\begin{frame}
	\titlepage
\end{frame}

%%%%%%%%%%%%%%%%%
\begin{frame}
	\frametitle{Outline} 

	\begin{itemize}
		\item Background.
		\item Progress update with results.
		\item Progress assessment.
		\item Timeline for completion.	
		\item Discussion of next steps.
	\end{itemize}
\end{frame}

%%%%%%%%%%%%%%%%


\begin{frame}
	\frametitle{What is \textit{Trichoderma}?}
	\begin{itemize}
		\item \textit{Trichoderma} is a genus of filamentous that is ubiquitous in soil and plays a significant role in nutrient cycling.
		\item They are also used in biocontrol and as biofertilizers.
		\item Known for their production of plant cell wall degrading enzymes, and \textbf{secondary metabolites}.
		\item Further genomic studies can help in understanding their biology and potential applications.
	\end{itemize}
	\vspace{0.05cm}
	\centering
	%\includegraphics[width=0.2\textwidth]{./Trichoderma_harzianum.jpg}

	\begin{figure}
		\centering
		\begin{minipage}{0.5\textwidth}
			\centering
			\includegraphics[width=0.35\linewidth]{./Trichoderma_harzianum.jpg}
			\caption{\textit{T. harzianum}}
		\end{minipage}\hfill
		\begin{minipage}{0.48\textwidth}
			\centering
			\includegraphics[width=0.6\linewidth]{./Trichoderma_petri.jpg}
			\caption{\textit{Trichoderma} colony}
		\end{minipage}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{Novel \textit{Trichoderma} Genomes}
	\begin{itemize}
		\item \textit{Trichoderma} species are diverse, with many species not yet fully characterized.
		\item DC1 and Tsth20, shown to improve drought and salt tolerance when applied to crops, and have been shown to breakdown hydrocarbons in soils.
		\item Recent advances in sequencing technology have made it possible to generate high-quality genomes for these species.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Motivation} 
	\begin{itemize}
		\item To better understand the biological mechanisms at work in \textit{Trichoderma} spp., we first need to understand their genomes by identifying genes within them.
		\item Various tools exist for gene prediction, but their implementations and performance can vary significantly.
		\item \textbf{Few studies have compared these tools in fungi, and even fewer in \textit{Trichoderma}}.
		\item In addition, increased accessibility to high-quality sequencing services has led to the generation of many new \textit{Trichoderma} genomes, including DC1 and Tsth20.
		\item These genomes provide an opportunity to evaluate gene finding tools in a comparative context, and contribute to the understanding of \textit{Trichoderma} biology. 
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Research Objectives}
	\begin{itemize}
		\item Assemble and evaluate novel assemblies of DC1 and Tsth20.
		\item Apply gene finders Braker2 and GeneMark in DC1, Tsth20, and three other RefSeq assemblies - \textit{T. reesei, T.harzianum} and \textit{T. virens}.
		\item Compare gene finding tools based on relevant criteria, including:
		\begin{itemize}
			 \item Proportions of gene lengths predicted.
			 \item Presence of functional domains and closely related protein sequences.
			 \item Presence of genes in AT-rich sequence.
			 \item Agreement of gene finders on start and stop positions of a gene. 
		\end{itemize}
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Workflow Overview}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/workflow-simple.pdf}
\end{frame}

\begin{frame}
	\frametitle{Assembly Results}
	\centering
	\vspace{0.5cm}
	\resizebox{\linewidth}{!}{
    \begin{tabular}{|c|c|c|c|c|c|c|}
		\hline
  		Name & Total Contigs & Total Length & Largest Contig & GC\% & N50 & L50 \\ \hline
     	DC1 & \cellcolor{green}8 & 38.6 Mb & \cellcolor{green}11.49 Mb & 47.97 & \cellcolor{green}5.69 Mb & \cellcolor{green}3 \\ \hline
      	Tsth20 & \cellcolor{green}7 & 41.58 Mb & \cellcolor{green}8.02 Mb & 47.33 & \cellcolor{green}6.52 Mb & \cellcolor{green}3 \\ \hline
      	\textit{T. harzianum} & 532 & 40.98 Mb & 4.08 Mb & 47.61 & 2.41 Mb & 7 \\ \hline
      	\textit{T. virens} & 93 & 39.02 Mb & 3.45 Mb & 49.25 & 1.83 Mb & 8 \\ \hline
      	\textit{T. reesei} & 77 & 33.39 Mb & 3.75 Mb & 52.82 & 1.21 Mb & 9 \\ \hline
    \end{tabular}}
	\begin{itemize}
		\item DC1 and Tsth20 assemblies are of high quality, with few contigs in comparison to other \textit{Trichoderma} assemblies.
		\item Input sequences and assemblies show a bimodal distribution of GC content.
		\item AT-rich sequence content may be related to transposable elements and repeat-induced mutations, which may be of interest in secondary metabolite production.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Gene Finding Results}
	\centering
	\resizebox{\linewidth}{!}{
	\begin{tabular}{|c|c|c|c|c|c|c|}
    	\hline
    	& \multicolumn{2}{c|}{Braker2} & \multicolumn{2}{c|}{GeneMark} & \multicolumn{2}{c|}{RefSeq} \\ \hline
     	Assembly & Genes & CDS & Genes & CDS & Genes & CDS \\ \hline
    	DC1 & 8546 & 8637 & 11353 & 11353 & N/A & N/A \\ \hline
    	Tsth20 & 8784 & 8858 & 12362 & 12362 & N/A & N/A \\ \hline
    	\textit{T. reesei} & \cellcolor{green}9659 & \cellcolor{green}10175 & 9196 & 9196 & 9109 & 9118 \\ \hline
    	\textit{T. harzianum} & 8314 & 8385 & 12164 & 12164 & \cellcolor{green}14269 & \cellcolor{green}14090 \\ \hline
    	\textit{T. virens} & 7801 & 7863 & 11866 & 11866 & 12405 & 12406 \\ \hline
  	\end{tabular}}
	\begin{itemize}
		\item Braker2 predicts more genes and coding sequences than GeneMark and RefSeq in \textit{T. ressei}, but fewer in other assemblies.
		\item GeneMark and RefSeq predictions are similar, except in \textit{T. harzianum}, where RefSeq predicts 17\% more genes.
		\item GeneMark only predicts one coding sequence per gene, while Braker2 and RefSeq predict multiple coding sequences.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Examining Coding Sequence Lengths}
	\centering
	\vspace{0.3cm}
	\resizebox{0.9\linewidth}{!}{
	\begin{tabular}{|c|c|c|c|c|c|c|}
      \hline
      Genome & Tool \#1 & Tool \#2 & \textit{P}-value  \\ \hline
      DC1 & Braker2 & GeneMark & $0.999$ \\ \hline
      Tsth20 & Braker2 & GeneMark & $0.965$ \\ \hline
      %\textit{T. reesei} & Braker2 & GeneMark & $9.481*10^{-07}$ \\ \hline
	  \textit{T. reesei} & \cellcolor{yellow}Braker2 & GeneMark & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
      \textit{T. reesei} & GeneMark & RefSeq & $0.002$ \\ \hline
      \textit{T. reesei} & \cellcolor{yellow}Braker2 & RefSeq & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
      %\textit{T. reesei} & Braker2 & RefSeq & $1.340*10^{-07}$ \\ \hline
	  \textit{T. harzianum} & Braker2 & GeneMark & $0.863$ \\ \hline
      %\textit{T. harzianum} & GeneMark & RefSeq & $4.313*10^{-52}$ \\ \hline
      \textit{T. harzianum} & GeneMark & \cellcolor{yellow}RefSeq & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
	  %\textit{T. harzianum} & Braker2 & RefSeq & $4.674*10^{-55}$ \\ \hline
	  \textit{T. harzianum} & Braker2 & \cellcolor{yellow}RefSeq & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
      \textit{T. virens} & Braker2 & GeneMark & $0.635$ \\ \hline
      %\textit{T. virens} & GeneMark & RefSeq & $7.352*10^{-12}$ \\ \hline
	  \textit{T. virens} & GeneMark & \cellcolor{yellow}RefSeq & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
	  %\textit{T. virens} & Braker2 & RefSeq & $1.794*10^{-09}$ \\ \hline
      \textit{T. virens} & Braker2 & \cellcolor{yellow}RefSeq & \cellcolor{yellow}$\textit{P}<0.005$ \\ \hline
    \end{tabular}}
\end{frame}


\begin{frame}
	\frametitle{BUSCO Results}
	\centering
	\includegraphics[width=0.65\textwidth]{../../working-thesis/figures/busco-screenshot.png}
\end{frame}

\begin{frame}
	\frametitle{Agreement of Gene Predictions}
	\centering
	\begin{itemize}
		\item In DC1 and Tsth20, Braker2 and GeneMark tend to agree on the start and stop positions of genes when they predict the same gene. Both gene finders have a large portion of singleton predictions.
	\end{itemize}
	\begin{figure}
		\includegraphics[width=0.9\textwidth]{../../working-thesis/figures/dc1-region-breakdown.png}
		\caption{DC1}
	\end{figure}
\end{frame}

\begin{frame}
	\vspace{1cm}
	\centering
	\begin{itemize}
		\item In \textit{T. reesei}, Braker2, GeneMark and RefSeq tend to disagree more on the start and stop positions of genes when they predict the same gene.
		\item Disagreement is more pronounced in \textit{T. harzianum} and \textit{T. virens}, where there are fewer genes with supportinng predictions from each gene finder.
		
	\end{itemize}
	\begin{figure}
		\includegraphics[width=0.8\textwidth]{../../working-thesis/figures/t-reesei-region-breakdown.png}
		\caption{\textit{T. reesei}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{InterProScan Results}
	\centering
	\vspace{0.35cm}
	\resizebox{\linewidth}{!}{
	\begin{tabular}{|c|c|c|c|c|c|c|}
    \hline
    Assembly & Braker2 & GeneMark & RefSeq \\ \hline
DC1 & ${73\%}$ & ${74.1\%}$ & \
N/A \\ \hline
    Tsth20 & ${73.3\%}$ & ${74.1\%}$
 & N/A \\ \hline
    \textit{T. reesei} & ${72.4\%}$ & ${76\%}$ & ${76.4\%}$ \\ \hline
    \textit{T. harzianum} & ${73.8\%}$ & ${74.5\%}$ & \cellcolor{yellow}${66.11\%}$ \\ \hline
    \textit{T. virens} & ${74.7\%}$ & ${74.8\%}$ & ${73.2\%}$ \\ \hline
  	\end{tabular}}
  	\begin{itemize}
		\item All three gene finders predict a similar proportion of genes with functional domains, except in the case of \textit{T. harzianum}, where RefSeq predicts a significantly lower proportion of genes with functional domains.
	\end{itemize}
\end{frame}

%\begin{frame}
%	\frametitle{BLAST Results}
%	\centering
%	\vspace{0.7cm}
%	\resizebox{1\linewidth}{!}{
%	\begin{tabular}{|c|c|c|c|c|c|c|}
%    	\hline
%   		Reference & Ref. Proteins & DC1 & Tsth20 & \textit{T. reesei} & \textit{T. harzianum} & \textit{T. virens}  \\ \hline
%    	\textit{Trichoderma atroviride} & 11807 & 11552 & 11080 & 10601 & 11081 & 11078 \\ \hline 
%    	\textit{Fusarium granminarium} & 13312 & 10327 & 10429 & 10064 & 10434 & 10490 \\ \hline
%    	\textit{Saccharomyces cerevisiae} & 6014 & 3537 & 3517 & 3445 & 3509 & 3500 \\ \hline
%  	\end{tabular}}
%	\vspace{0.2cm}
%	\begin{itemize}
%		\item The \textit{T. atroviride} and \textit{Fusarium} datasets are well respresented in the tblastn searches, while the \textit{S. cerevisiae} dataset is less well represented.
%	\end{itemize}
%\end{frame}

\begin{frame}
	\frametitle{tblastn Results}
	\resizebox{\linewidth}{!}{
	\begin{tabular}{|c|c|c|c|c|}
    \hline
    Subject & Query & Braker2 & GeneMark & RefSeq \\ \hline
    DC1 & \textit{T. atroviride} & 5902 & \cellcolor{yellow}4679 & N/A \\ \hline
    DC1 & \textit{F. graminarium} & 4955 & \cellcolor{yellow}4114 & N/A \\ \hline
    DC1 & \textit{S. cerevisiae} & 2105 & \cellcolor{yellow}1850 & N/A \\ \hline
	\textit{T. reesei} & \textit{T. atroviride} & 5072 & \cellcolor{green}5174 & 4989 \\ \hline
    \textit{T. reesei} & \textit{F. graminarium} & 4577 & \cellcolor{green}4685 & 4529 \\ \hline
    \textit{T. reesei} & \textit{S. cerevisiae} & 2055 & \cellcolor{green}2114 & 2022 \\ \hline
	\textit{T. harzianum} & \textit{T. atroviride} & 6363 & \cellcolor{yellow}4611 & 6835 \\ \hline
    \textit{T. harzianum} & \textit{F. graminarium} & 5659 & \cellcolor{yellow}4198 & 5982 \\ \hline
    \textit{T. harzianum} & \textit{S. cerevisiae} & 2424 & \cellcolor{yellow}1963 & 2560 \\ \hline
	\end{tabular}}
	\begin{itemize}
		\item Regions with Braker2 and RefSeq gene predictions consistently have more tblastn hits than GeneMark, with the exception of \textit{T. reesei}.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Gene Predictions in AT-rich Sequence}
	\centering
	\resizebox{0.7\linewidth}{!}{
    \begin{tabular}{|c|c|c|c|}
      \hline
      Assembly & Braker2 & GeneMark & RefSeq \\ \hline
      DC1 & \cellcolor{yellow}31 & \cellcolor{yellow}11 & N/A \\ \hline
      Tsth20 & \cellcolor{yellow}11 & \cellcolor{yellow}2 & N/A \\ \hline
      \textit{T. reesei} & 39 & 48 & 107 \\ \hline
      \textit{T. harzianum} & 81 & 30 & 154 \\ \hline
      \textit{T.virens} & \cellcolor{yellow}21 & \cellcolor{yellow}8 & \cellcolor{yellow}20 \\ \hline
    \end{tabular}}
	\vspace{0.3cm}
	\begin{itemize}
		\item Very few genes are predicted in AT-rich regions of DC1, Tsth20 and \textit{T. virens} assemblies in comparison to the \textit{T. reesei} and \textit{T. harzianum} assemblies.
		\item Possibly due to higher quality assemblies? Although \textit{T. virens} is still fragmented.
		\item A two-sided binomial test confirms that gene finders do not predict the same proportion of genes in AT-rich sequence as they do in normal genomic sequence. 
	\end{itemize}
\end{frame}

%\begin{frame}
%	\centering
%	\vspace{2cm}
%	\resizebox{\linewidth}{!}{
%	\begin{tabular}{|c|c|c|c|c|c|}
%      	\hline
%      	Tool & DC1 & Tsth20 & \textit{T. reesei} & \textit{T. harzianum} & \textit{T. virens} \\ \hline
%     	 Braker2 & $9.56*10^{-181}$ & $1.14*10^{-259}$ & $2.68*10^{-96}$ & $4.05*10^{-140}$ & $1.35*10^{-35}$ \\ \hline
%      	GeneMark & $5.12*10^{-216}$ & $0.0$ & $5.66*10^{-49}$ & $5.37*10^{-219}$ & $5.31*10^{-35}$ \\ \hline
%      	RefSeq & N/A & N/A & $1.29*10^{-49}$ & $2.44*10^{-205}$ & $7.40*10^{-33}$ \\ \hline
%    \end{tabular}}
%	\begin{itemize}
%		\item The results of the two-sided binomial test indicate that the number of genes predicted in AT-rich genomic sequence is not proportional to the fraction of genomic sequence they comprise.
%	\end{itemize}
%\end{frame}

\begin{frame}
	\frametitle{Conclusions}
	\begin{itemize}
		\item In terms of gene finding performance based on several criteria, RefSeq predictions are typically the best, while GeneMark performs the worst.
		\item Braker2 performs well in \textit{T. reesei}, but not as well in the other assemblies. Users should be careful with the training data selected for Braker2, as it can have a significant impact on the results.
		\item RefSeq predictions are not always available, so Braker2 can be used if appropriate training is available, but GeneMark can be used as a fallback.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Future Work}
	\begin{itemize}
		\item \textbf{Explore the landscape of secondary metabolite gene clusters in \textit{Trichoderma} genomes further.}
		\item Extend the analysis to include more \textit{Trichoderma} species, more gene finding tools, and more datasets.
		\item Investigate further methods for validation of gene predictions.
		\item Continue to analyze the newly assembled genomes of DC1 and Tsth20.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Timeline for Completion}
	\begin{itemize}
    	\item Initial draft of thesis complete - end of August 2025
    	\item Revisions and changes - Sept./Oct. 2025
    	\item Submit thesis to committee - Early November 2025
    	\item Schedule defence for December 2025 or January 2026
    	\item Time for extra revisions post-defence - January/early February 2026
    	\item Completion - End of February 2026
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Current Status}
	\begin{itemize}
		\item Introduction - 70\%
		\item Background - 85\%
		\item Methods - 90\%
		\item Research Questions - 85\%
		\item Results and Discussion - 80(?)\%
		\item Conclusions and Future Work - 75\%
		\item What to focus on next?
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Acknowledgements}
	\begin{itemize}
		\item My supervisors for their committed support through COVID and other challenges.
		\item The Global Institute for Food Security for providing the data for this project as well as a portion of my funding.
		\item My committee members for their feedback and support.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Datasets and Tools}
	\begin{itemize}
		\item \textbf{Datasets:}
		\begin{itemize}
			\item Novel \textit{Trichoderma} genomes: DC1 and Tsth20.
			\item Reference genomes and annotations: \textit{T. reesei, T. harzianum, T.virens}.
			\item RNAseq training data from \textit{T. reesei}. 
			\item Benchmarking Universal Single-Copy Orthologs (BUSCO) fungal database.
			\item Protein sequence queries for tblastn from \textit{T. atroviride, Fusarium graminearum, and Saccharomyces cerevisiae}.	
		\end{itemize}
		\item \textbf{Tools:}
		\begin{itemize}
			\item Sequence processing: FastQC, Trimmomatic, Hisat2.
			\item Genome assembly: NextDenovo and NextPolish.
			\item Gene finding tools: Braker2 and GeneMark-ES.
			\item Evaluation tools: BUSCO, tblastn, InterProScan, and custom scripts.
		\end{itemize}
	\end{itemize}
\end{frame}

\begin{frame}
	\vspace{0.75cm}
	\centering
	\includegraphics[width=0.5\textwidth]{../../working-thesis/figures/conclusion-snip.png}
\end{frame}

\begin{frame}
	\frametitle{Image Credits}
	\begin{itemize}
		\item \textit{T. harzianum} image: \url{https://en.wikipedia.org/wiki/Trichoderma}
		\item \textit{Trichoderma colony} image:\url{https://biocontrol.entomology.cornell.edu/pathogens/trichoderma.php}
	\end{itemize}
\end{frame}

\end{document}