\documentclass[t]{beamer}
\usepackage{lmodern} % Need to use this package for proper font encoding.
\usepackage[T1]{fontenc}
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage{multicol}
\usepackage{booktabs}
\usepackage{bookmark}
\usepackage[table]{xcolor}
% Title slide information
\title{Thesis Defence: Evaluation of Gene Finding Tools When Applied to \textit{Trichoderma} Genomes}
\date{January, 2026} 
\author{Connor Burbridge}
\titlegraphic{UsaskAerial.jpg} 
\institute{Department of Computer Science} 
\usetheme{usask}

\begin{document}

\begin{frame}
	\titlepage
\end{frame}

%%%%%%%%%%%%%%%%%
\begin{frame}
	\frametitle{Outline}
	\vspace{0.7cm} 
    \Large
	\begin{itemize}
		\item \textbf{\textit{Trichoderma} fungi.}
		\item Motivation.
		\item Research objectives.
		\item Workflow overview.
		\item Results highlights.
		\item Conclusions and future work.
	\end{itemize}
\end{frame}

%%%%%%%%%%%%%%%%


\begin{frame}
	\frametitle{What is \textit{Trichoderma}?}
	\begin{itemize}
		\item Filamentous fungi common in soil; key role in nutrient cycling.
		\item Used in biocontrol and as biofertilizers.
		\item Produce plant cell wall degrading enzymes and \textbf{secondary metabolites}.
		\item Future studies reveal their biology and potential applications.
	\end{itemize}
	\vspace{0.05cm}
	\centering
	%\includegraphics[width=0.2\textwidth]{./Trichoderma_harzianum.jpg}

	\begin{figure}
		\centering
		\setbeamerfont{footnote}{size=\tiny}
		\begin{minipage}{0.5\textwidth}
			\centering
			\includegraphics[width=0.35\linewidth]{./Trichoderma_harzianum.jpg}
			\caption{\textit{T. harzianum~\footnotemark}}
		\end{minipage}\hfill
		\begin{minipage}{0.48\textwidth}
			\centering
			\includegraphics[width=0.6\linewidth]{./Trichoderma_petri.jpg}
			\caption{\textit{Trichoderma} colony~\footnotemark}
		\end{minipage}
		\footnotetext[1]{Source: \url{https://en.wikipedia.org/wiki/Trichoderma}}
		\footnotetext[2]{Source: \url{https://biocontrol.entomology.cornell.edu/pathogens/trichoderma.php}}
	\end{figure}
\end{frame}

\begin{frame}
	\frametitle{Novel \textit{Trichoderma} Genomes}
	\begin{itemize}
		\item \textit{Trichoderma} species are diverse, with many species not yet fully characterized.
		\item DC1 and Tsth20 both improve salt and drought tolerance.
		\item Tsth20 shows potential as a bioremediation agent.
		\item Recent advances in sequencing technology have made it possible to generate high-quality genomes for these species.
		\item Joint effort by Brendan Ashby, the Kaminskyj lab, and the Global Institute for Food Security.
	\end{itemize}
\end{frame}

\begin{frame}
    \Large
	\vspace{2cm}
	\begin{itemize}
		\item \textcolor{gray}{\textit{Trichoderma} fungi.}
		\item \textbf{Motivation.}
		\item Research objectives.
		\item Workflow overview.
		\item Results highlights.
		\item Conclusions and future work.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Motivation} 
	\begin{itemize}
		\item To understand \textit{Trichoderma} biology, we must identify genes within their genomes.
		\item Gene prediction tools vary significantly in implementation and performance.
		\item Novel genomes with novel biology require gene annotation to connect phenotypes to genotypes.
		\item New high-quality \textit{Trichoderma} genomes (DC1 and Tsth20) enable this comparative evaluation of gene finding tools. 
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Gene Finding is Still Challenging}
	\begin{itemize}
		\item We've been finding genes for decades. What's going on?
		\item Exons and introns as well as start and stop positions make gene finding tricky.
		\item Inherent properties of genomes also complicate gene finding.
		\item Stop codons in AT-rich regions are highly prevalent.
	\end{itemize}
	\begin{columns}
		\begin{column}{0.4\textwidth}
			\centering
			\includegraphics[scale=0.175]{../../working-thesis/figures/dc1-low-gc-fastqc.png}
		\end{column}
		\begin{column}{0.5\textwidth}
			\includegraphics[scale=0.33]{../../working-thesis/figures/gc-plot.pdf}
		\end{column}
		
	\end{columns}
\end{frame}

\begin{frame}
	\frametitle{The Details Matter}
	\begin{itemize}
		\item Minor changes in start/stop positions can have major impacts on predicted protein sequences.
		\item Ex. The subcellular localization process relies heavily on N-terminal sequences. 
		\item Links back to cell wall degrading enzymes and secondary metabolites.
	\end{itemize}
	\vspace{0.3cm}
	\centering
	\includegraphics[width=0.5\textwidth]{./subcellular-localization.jpg}
\end{frame}

\begin{frame}
	\frametitle{Variability in Gene Predictions}
	\centering
	\includegraphics[width=0.7\textwidth]{../../working-thesis/figures/igv/igv-complicated-thin.png}
	\includegraphics[width=0.7\textwidth]{../../working-thesis/figures/igv/ips-model-disagree.png}
\end{frame}

\begin{frame}
	\Large
	\vspace{2cm}
	\begin{itemize}
		\item \textcolor{gray}{\textit{Trichoderma} fungi.}
		\item \textcolor{gray}{Motivation.}
		\item \textbf{Research objectives.}
		\item Workflow overview.
		\item Results highlights.
		\item Conclusions and future work.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Research Objectives}
	\begin{itemize}
		\item Assemble and evaluate novel assemblies of DC1 and Tsth20.
		\item Apply gene finders Braker2 and GeneMark in DC1, Tsth20, and three other RefSeq assemblies.
		\item Compare gene finding tools based on relevant criteria, including:
		\begin{itemize}
			 \item Number of genes and isoforms predicted.
			 \item Presence of conserved single-copy orthologs (BUSCOs).
			 \item Presence of functional domains and closely related protein sequences.
			 \item Presence of genes in AT-rich sequence.
			 \item Agreement of gene finders on start and stop positions of a gene. 
		\end{itemize}
	\end{itemize}
\end{frame}

\begin{frame}
	\Large
	\vspace{2cm}
	\begin{itemize}
		\item \textcolor{gray}{\textit{Trichoderma} fungi.}
		\item \textcolor{gray}{Motivation.}
		\item \textcolor{gray}{Research objectives.}
		\item \textbf{Workflow overview.}
		\item Results highlights.
		\item Conclusions and future work.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Workflow Overview}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/workflow-simple.pdf}
\end{frame}

\begin{frame}
	\frametitle{Region Identification}
	\centering
	\includegraphics[width=0.65\textwidth]{../../working-thesis/figures/igv/igv-agreement.png}
	\vspace{1cm}
	\includegraphics[width=0.7\textwidth]{../../working-thesis/figures/igv/igv-complicated-thin.png}
\end{frame}

\begin{frame}
	\Large
	\vspace{2cm}
	\begin{itemize}
		\item \textcolor{gray}{\textit{Trichoderma} fungi.}
		\item \textcolor{gray}{Motivation.}
		\item \textcolor{gray}{Research objectives.}
		\item \textcolor{gray}{Workflow overview.}
		\item \textbf{Results highlights.}
		\item Conclusions and future work.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Assemblies of DC1 and Tsth20}
	\begin{itemize}
		\item Both assemblies are of high quality.
		\item Few contigs, which is good for gene finding.
		\item BUSCO analysis sees similar benefits.
		\item RefSeq assemblies are highly fragmented in comparison.
	\end{itemize}
	\centering
	\vspace{0.6cm}
	\includegraphics[width=\textwidth]{../../working-thesis/figures/assembly-stats-snip.png}
\end{frame}

\begin{frame}
	\frametitle{Small but Significant Details}
	\begin{itemize}
		\item Before diving too deep, how can we objectively confirm that gene finding tools are systematically different?
		\item Gene finders may be biased to predict certain gene lengths.
		\item Kolmogorov-Smirnov (KS) tests were performed on gene length distributions from preliminary gene predictions.
		\begin{center}
			\includegraphics[width=0.9\textwidth]{../../working-thesis/figures/t-reesei-cdf-lengths-log.pdf}
		\end{center}
	\end{itemize}	
\end{frame}

\begin{frame}
	\frametitle{Regions of Gene Predictions}
	\begin{itemize}
		\item Agreement of gene finders assessed by identifying regions of gene predictions.
		\item Regions classified based on agreement of start and stop positions.
		\item Transitive closure used to group overlapping predictions.
	\end{itemize}
	\centering
	\includegraphics[width=0.9\textwidth]{../../working-thesis/figures/t-reesei-region-breakdown.png}
\end{frame}

\begin{frame}
	\frametitle{AT-rich Sequence Analysis}
	\begin{itemize}
		\item GC content varies across genomes.
		\item Gene finders struggle in AT-rich (>72\% AT content) regions.
	\end{itemize}
	\centering
	\includegraphics[width=0.9\textwidth]{../../working-thesis/figures/gc-percent-snip.png}
	\vspace{0.2cm}
	\includegraphics[width=0.45\textwidth]{../../working-thesis/figures/gc-gene-counts-snip.png}
\end{frame}
 
\begin{frame}
	\frametitle{Secondary Metabolite Gene Clusters}
	\begin{itemize}
		\item antiSMASH used to identify candidate secondary metabolite gene clusters.
		\item Tsth20 contains multiple candidate clusters.
		\item 35 candidates from Braker2 predictions and 58 from GeneMark.
		\item Limited exploration of predicted clusters performed.
	\end{itemize}
	\centering
	\vspace{0.2cm}
	\includegraphics[width=0.8\textwidth]{../../working-thesis/figures/braker-antismash-tsth20.png}
\end{frame}

\begin{frame}
	\frametitle{Ranking Gene Finders}
	\vspace{0.5cm}
	\centering
	\includegraphics[width=0.9\textwidth]{../../working-thesis/figures/conclusion-snip.png}
\end{frame}

\begin{frame}
	\Large
	\vspace{2cm}
	\begin{itemize}
		\item \textcolor{gray}{\textit{Trichoderma} fungi.}
		\item \textcolor{gray}{Motivation.}
		\item \textcolor{gray}{Research objectives.}
		\item \textcolor{gray}{Workflow overview.}
		\item \textcolor{gray}{Results highlights.}
		\item \textbf{Conclusions and future work.}
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Key Findings}
	\begin{itemize}
		\item Overall performance of gene finders are similar, but vary by specific criteria.
		\item Braker2 ranks highest, particularly when relevant training data is available.
		\item Candidate secondary metabolite gene clusters were found in Tsth20 assembly.
		\item DC1 and Tsth20 assemblies are of high quality, providing a solid foundation for future genomic studies.
		\item Explored a reference-agnostic approach to gene prediction comparisons.
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Future Work}
	\vspace{0.6cm}
	\begin{itemize}
		\item A few of many potential avenues...
		\begin{itemize}
			\item Explore additional gene finding tools.
			\item Adjust parameters of existing tools.
			\item Experiment with different types of training data for Braker2.
			\item Extend analysis of identified regions of gene predictions.
			\item Explore secondary metabolite gene clusters in more detail.
			\item Expand the number of \textit{Trichoderma} genomes analyzed.
		\end{itemize}
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Acknowledgements}
	\begin{itemize}
		\item My supervisors for their committed support through COVID and other challenges.
		\item Brendan Ashby and the Kaminskyj lab for providing the data for this project.  
		\item Global Institute for Food Security for providing a portion of my funding.
		\item My committee members for their feedback and support.
	\end{itemize}
\end{frame}

\begin{frame}
	\vspace{3cm}
	\begin{center}
		\Huge{Questions and Discussion}
	\end{center}
\end{frame}

\begin{frame}
	\frametitle{Workflow Details: Assembly}
	\centering
	\includegraphics[width=0.6\textwidth]{../../working-thesis/figures/assembly-met.pdf}
\end{frame}

\begin{frame}
	\frametitle{Workflow Details: Gene Finding}
	\centering
	\includegraphics[width=0.78\textwidth]{../../working-thesis/figures/gene-finding-met.pdf}
\end{frame}

\begin{frame}
	\frametitle{Workflow Details: Evaluation}
	\centering
	\includegraphics[width=01\textwidth]{../../working-thesis/figures/eval-met.pdf}
\end{frame}

\begin{frame}
	\frametitle{Workflow Details: Regions}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/region-workflow.pdf}
\end{frame}

\begin{frame}
	\frametitle{Datasets and Tools}
	\begin{itemize}
		\item \textbf{Datasets:}
		\begin{itemize}
			\item Novel \textit{Trichoderma} genomes: DC1 and Tsth20.
			\item Reference genomes and annotations: \textit{T. reesei, T. harzianum, T.virens}.
			\item RNAseq training data from \textit{T. reesei}. 
			\item Benchmarking Universal Single-Copy Orthologs (BUSCO) fungal database.
			\item Protein sequence queries for tblastn from \textit{T. atroviride, Fusarium graminearum, and Saccharomyces cerevisiae}.	
		\end{itemize}
		\item \textbf{Tools:}
		\begin{itemize}
			\item Sequence processing: FastQC, Trimmomatic, Hisat2.
			\item Genome assembly: NextDenovo and NextPolish.
			\item Gene finding tools: Braker2 and GeneMark-ES.
			\item Evaluation tools: BUSCO, tblastn, InterProScan, and custom scripts.
		\end{itemize}
	\end{itemize}
\end{frame}

\begin{frame}
	\frametitle{Phylogenetics}
	\centering
	\includegraphics[width=0.72\textwidth]{../../working-thesis/figures/trichoderma-phylogeny-few-outgroups.pdf}
\end{frame}

\begin{frame}
	\frametitle{Region Identification Algorithm}
	\centering
	\includegraphics[width=0.55\textwidth]{../../working-thesis/figures/pseudocode-snapshot.png}
\end{frame}

\begin{frame}
	\frametitle{BUSCO Results}
	\vspace{0.2cm}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/busco-complete-counts.pdf}
\end{frame}

\begin{frame}
	\frametitle{InterProScan Results}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/interproscan-barplot.pdf}
\end{frame}

\begin{frame}
	\frametitle{BLAST Results}
	\centering
	\includegraphics[width=\textwidth]{../../working-thesis/figures/blast-tatroviride.pdf}
\end{frame}

\begin{frame}
	\frametitle{AT-rich Sequence Localization}
	\centering
	\vspace{0.3cm}
	\includegraphics[width=\textwidth]{../../working-thesis/figures/dc1-atRich-windows.png}
	\includegraphics[width=\textwidth]{../../working-thesis/figures/t-reesei-atRich-window-snapshot.png}
\end{frame}

\end{document}