\documentclass{uofsthesis-cs}
\usepackage[noblocks]{authblk}
\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}
\usepackage{titlesec}
\usepackage{indentfirst}
\usepackage{hyperref}
\usepackage{booktabs}
\usepackage{makecell}
\usepackage{arydshln}
\usepackage{tikz}
\usepackage{algpseudocode}
\usepackage{algorithm}
\usepackage{textcomp}

\def\checkmark{\tikz\fill[scale=0.4](0,.35) -- (.25,0) -- (1,.7) -- (.25,.15) -- cycle;}

\title{Comparative analysis of Gene Finding tools when applied to
  \textit{Trichoderma} genomes} 
  \author{Connor Burbridge}

\parindent=14pt

\abstract{
This thesis presents a comparative analysis of gene prediction tools applied to genomes of the genus \textit{Trichoderma}. The study evaluates the performance of several widely used gene finding algorithms, assessing their accuracy, sensitivity, and specificity using both computational benchmarks and biological validation. Genomic datasets from multiple \textit{Trichoderma} species were analyzed, and the predicted gene models were compared against reference annotations and experimental data. The results highlight the strengths and limitations of each tool, providing recommendations for optimal gene annotation strategies in fungal genomics. Briefly summarizing the results, we found that gene finders generally showed high sensitivity in detecting conserved genes, and the predicted gene models contained functional domains as validated by InterProScan. However, discrepancies were observed among the tools in start and stop positions, exon-intron structures, and alternative splicing events. Notably, gene prediction accuracy declined in AT-rich genomic regions, where few genes were identified. 

When selecting a gene finding tool, users should consider available annotations, genomic context, and training data. Braker2 performed best overall when relevant high-quality training data was available. RefSeq ranked second but performed poorly in BUSCO and InterProScan categories. GeneMark performed worst overall but still provides useful gene predictions when other options are not available. From these results, we make the following recommendations when selecting a gene finding tool for \textit{Trichoderma} genomes: use Braker2 when high-quality training data exists; use RefSeq when training data is unavailable but RefSeq annotations exist; otherwise use GeneMark. Users should critically evaluate RefSeq annotations, as underlying assembly quality may impact predictions.

In addition to the main findings, the genomes of two novel \textit{Trichoderma} species, DC1 and Tsth20, were successfully assembled and annotated, resulting in high-quality genomic resources. The assemblies exhibited near-chromosomal scale contigs and performed well in standard assembly metrics. The gene predictions generated by the evaluated tools produced a rich dataset of annotations for these genomes, which can be leveraged for further biological investigations.

}

\acknowledgements{I would like to acknowledge both
  Dr. Kusalik and Dave Schneider for their excellent support and
  mentorship during both COVID and my MSc. project. I would also like
  to thank Brendan Ashby and Dr. Leon Kochian for providing both data
  and additional financial support. I also thank my family and girlfriend for their unwavering support and encouragement throughout my studies.}

\loa{
\abbrev{A}{Adenine}
\abbrev{BGC}{Biosynthetic gene cluster}
\abbrev{BLAST}{Basic local alignment search tool}
\abbrev{BUSCO}{Benchmarking universal single-copy orthologs}
\abbrev{C}{Cytosine}
\abbrev{CDF}{Cumulative distribution function}
\abbrev{CDS}{Coding sequence}
\abbrev{CPU}{Central processing unit}
\abbrev{DNA}{Deoxyribonucleic acid}
\abbrev{G}{Guanine}
\abbrev{GFF}{General feature format}
\abbrev{GC}{Guanine cytosine}
\abbrev{GMO}{Genetically modified organism}
\abbrev{HGT}{Horizontal gene transfer}
\abbrev{HMM}{Hidden Markov model}
\abbrev{HPC}{High performance computing}
\abbrev{Kb}{Kilobase\(s\)}
\abbrev{Mb}{Megabase\(s\)}
\abbrev{Mb}{Mega-base}
\abbrev{MITE}{Minature inverted-repeat transposable element}
\abbrev{NCBI}{National center for biotechnology information}
\abbrev{NGS}{Next generation sequencing}
\abbrev{PCR}{Polymerase chain reaction}
\abbrev{RIP}{Repeat-induced point (mutation)}
\abbrev{RNA}{Ribonucleic acid}
\abbrev{RSMI}{Root, soil and microbial interactions}
\abbrev{T}{Thymine}
\abbrev{TE}{Transposable element}
\abbrev{TDR}{Terminal direct repeat}
\abbrev{TIR}{Terminal inverted repeat}
\abbrev{UTR}{Untranslated region}
\abbrev{WGS}{Whole genome shotgun (sequencing)}
}


% From Yujie: Assume I am only working on chapter 2 for now, so only include that
% chapter in the main document. This is useful for working on a single
% chapter at a time, and speeds up compilation.
%\includeonly{chapter-2}


\begin{document}
\maketitle
\frontmatter


\include{intro}
\include{chapter-2}
\include{chapter-3}
\include{chapter-4}
\include{chapter-5}
\include{chapter-6}

\uofsbibliography[unsrt]{./refs/full-bib}

%\include{supplemental}

\end{document}
