\documentclass[subfigure]{uofsthesis-cs}
\usepackage[noblocks]{authblk}
\usepackage{graphicx}
\usepackage{subfigure}
\usepackage{titlesec}
\usepackage{indentfirst}
\usepackage{hyperref}

\title{Comparative analysis of Gene Finding tools when applied to
  \textit{Trichoderma} genomes} \author{Connor Burbridge} \affil{USask
  NSID: cbe453 \\ USask ID no.\ 11162928 \\ Supervisors: Dave
  Schneider \& Tony Kusalik\\}

\parindent=14pt

\abstract{}

\acknowledgements{O would like to acknoweldge both
  Dr. Kusalik and Dave Schneider for their excellent support and
  mentorship during both COVID and my MSc. project. I would also like
  to thank Brendan Ashby and Dr. Leon Kochian for providing both data
  and additional financial support.}


\begin{document}
\maketitle
\frontmatter

\include{chapter-1}

\include{chapter-2}

\include{chapter-3}

\section{Discussion}

\section{Conclusions}

\include{supplemental}

\clearpage

\section{References}
\begin{thebibliography}{99}
\bibitem{GMO} Zhang, C., Wohlhueter, R., Zhang, H. (2016). Genetically
  modified foods: A critical review of their promise and
  problems. \textit{Food Science and Human Wellness}, \textit{5}(3),
  116-123.

\bibitem{Trichoderma} Hermosa, R., Viterbo, A., Chet, I., Monte,
  E. (2012). Plant-beneficial effects of \textit{Trichoderma} and of
  its genes. \textit{Microbiology}. \textit{38},
  17-25. 

\bibitem{Secretome} Ramirez-Valdespino, C., Casas-Flores, S.,
  Olmedo-Monfil, V. (2019). \textit{Trichoderma} as a Model to Study
  Effector-Like Molecules. \textit{Frontiers in Microbiology}, \textit{15}.

\bibitem{Drought} Repas, T.S., Gillis, D.M., Boubakir, Z., Bao,
  Xioahui., Samuels, G.J., Kaminskyj, S.G.W. (2017). Growing plants on
  oily, nutrient-poor soil using a native symbitoic
  fungus. \textit{PLOS ONE}, \textit{12}(10).

\bibitem{Kaminskyj} Kaminskyj et al. (2012). Method for increasing
  plant growth using the fungus \textit{Trichoderma harzianum} (United
  States PCT/CA2O1O/OO1454). https://patentimages.storage.googleapis.com/57/4f/7f/5ff60c47045f01/US20120178624A1.pdf
  
\bibitem{Braker2} Hoff, K. J., Lomsadze, A., Borodovsky, M., Stanke,
  M. (2019). Whole-Genome Annotation with BRAKER. \textit{Methods Mol
    Biol.}, \textit{1962}, 65-95.

\bibitem{GeneMarkES} Borodovsky, M., Lomsadze, A. (2011). Eukaryotic
  Gene Prodeiction using GeneMark.hmm-E and
  GeneMark-ES. \textit{Current Protocols in
    Bioinformatics}. \textit{4}.

\bibitem{Glimmer} Majoros, W.H., Pertea, M., Salzberg, S.L. (2004) TigrScan and
  GlimmerHMM: two open-source \textit{ab initio} eukaryotic
  gene-finders. \textit{Bioinformatics}. \textit{20}(9), 2878-2879.
  
\bibitem{AUGUSTUS} Stanke, M., Morgenstern, B. (2005). AUGUSTUS: a web
  server for gene prediction in eukaryotes that allows user-defined
  constraints. \textit{Nucleic Acids Research}. \textit{33}, 465-467.
  
\bibitem{InterProScan} Jones, P., Binns, D., Chang, H. Y., Fraser,
  W., Li, W., Hunter, S. (2014). InterProScan 5: genome-scale
  protein function
  classification. \textit{Bioinformatics}. \textit{30}(9),
  1236-1240.

\bibitem{AnnotationSummary} Yandell, M., Ence, D. (2012). A beginner's
  guide to eukaryotic genome annotation. \textit{Nature Reviews
    Genetics}. \textit{13}, 329-342.
  
\bibitem{assembly} Sohn, J., Nam, J. (2016) The present and future of
  \textit{de novo} whole-genome assembly. \textit{Briefings in
    Bioinformatics}. \textit{19}1, 23-40.

\bibitem{spades} Bankevich, A., Nurk, S., Antipov, D., et
  al. (2012). SPAdes: a new genome assembly algorithm and its
  applications to single-cell sequencing. \textit{Joural of
    Computational Biology}. \textit{19}(5), 455-477. 
  
\bibitem{Repeats} Lerat., E. (2009). Identifying repeats and
  transposable elements in sequences genomes: how to find your way
  through the dense forest of programs. \textit{Nature
    Heredity}. \textit{104}, 520-533.

\bibitem{fungalrepeats} Li, WC., Huang, CH., Chen, CL. et
  al. (2017). Trichoderma reesei complete genome sequence,
  repeat-induced point mutation, and partitioning of CAZyme gene
  clusters. \textit{Biotechnol Biofuels}. \textit{10}, 170.

\bibitem{dontMask} Slotkin, R.K. (2018). The case for not masking away
  repetitive DNA. \textit{Mobile DNA}. \textit{9}(1), 15.
  
\bibitem{GeneFinding} Wang, Z., Chen, Yazhu., Li, Y. (2016). A Brief
  Review of Computational Gene Prediction Methods. \textit{Genomics
    Proteomics Bioinformatics}. \textit{2}4, 216-221.

\bibitem{RepeatMasker} Smit, A., Hubley, R., Green, P. (2013-2015)
  RepeatMasker Open-4.0. https://www.repeeatmasker.org

\bibitem{Infernal} Nawrocki, E.P., Eddy, S.R. (2013). Infernal 1.1:
  100-fold faster RNA homology
  searches. \textit{Bioinformatics}. \textit{29}, 2933-2935.

\bibitem{GenomeThreader} Gremme, G., Brendel, V., Sparks, M.E., Kurtz,
  S. (2005) Engineering a software tool for gene structure prediction
  in higher organisms. \textit{Information and Software
    Technology}. \textit{47}, 965-978.

\bibitem{Validation} Adav, S.S., Sze, S.K. (2014). Chapter 8 -
  \textit{TRichoderma} Secretome: An Overview. \textit{Biotechnology
    and Biology of Trichoderma}. 103-104. 
  
\end{thebibliography}
>>>>>>> b715090053dedb063b1eacf591f238315062ce83

\end{document}
