\section{Selection of Gene Finding Tool}\label{chapter:conclusion}

With all of these results, it makes sense to explore the question of
which gene finding tool one should choose for optimal gene prediction
performance in a given application. Comparisons drawn in this section are made in the context
of \textit{Trichoderma} assemblies and may not extend to other
datasets. Observations from the results portion of this work have been
converted to a ranking for each gene finder, relative to the other gene finders. Values for each category are ranked from 0 to 3, with 0 indicating that a feature was unavailable or not applicable, and ranks 1 through 3 when there are three gene finders being considered, but 1 or 2 whether there are two gene finders being considered, indicating performance relative to the other gene finder(s). Ties are allowed if the performance of two gene finders are similar. Categories considered include availability of the gene finding tool, ease of installation, ease of use, number of genes predicted, number of isoforms predicted, BUSCO performance, performance in AT-rich sequence, and predictions with InterProScan support.
Results for these categories are summarized in
Table~\ref{table:final-score}. We note that not all results are included in this table, most notably the region analysis in Section~\ref{section:regions}, as more exploration of those results is needed. The assembly results are also not included, as they are not reflective of the characteristics of the gene finders themselves.
In both the publicly available and non-publicly available scenarios, Braker2 comes out on top, with RefSeq coming in second when availability is not considered. While Braker2 does perform the best, it is important to note that its performance is highly dependent on the quality and relevance of the training data provided. RefSeq performs well across most categories, particularly when considering number of genes predicted, isoform prediction, and predictions in AT-rich sequence, but if the user's organism is not closely represented in RefSeq, it is not an option.
GeneMark comes in last, performing well in terms 
of genes predicted, BUSCO coverage, and Pfam support for the genes that it predicts. We do note that these criteria are not fully objective; however they do provide a useful framework for comparing the gene finders, and are based on observations presented in this work. 

Relating these observations to use-case scenarios requires consideration of the data available to the user. In the case that a user has access to RefSeq annotation data for their organism, the gene predictions will likely be of good quality, although if new training data is available for the organism, gene predictions from a trained Braker2 model will supplement the RefSeq predictions. We note that users should always investigate the quality of the RefSeq assemblies, as some assemblies may be of low quality despite their status as a RefSeq genome assembly, which could affect gene finding performance. In the case that the organism of interest is not a RefSeq organism and no RefSeq annotation is available, Braker2 is the best option, provided that relevant training data can be found. If no training data is available, GeneMark is a reasonable choice, and although its performance is not as good as the other gene finders, it still predicts a useful set of genes. 

\begin{table}
  \centering
  \begin{tabular}{|c|c|c|c|}
    \hline
    Category & Braker2 & GeneMark & RefSeq \\ \hline
    Availability & 2 & 2 & 0 \\ \hline
    Ease of install & 1 & 2 & 0 \\ \hline
    Ease of use & 2 & 2 & 0 \\ \hdashline
    Number of genes predicted & 1 & 3 & 3 \\ \hline
    Number of isoforms predicted & 3 & 0 & 2 \\ \hline
    BUSCO Performance & 3 & 2 & 1 \\ \hline
    Performance in AT-rich sequence & 2 & 1 & 3 \\ \hline
    Predictions with InterProScan support & 2 & 3 & 1 \\ \hline
    Cumulative Rank (Considering Availability) & 16 & 15 & N/A \\ \hline
    Cumulative Rank (Ignoring Availability) & 11 & 9 & 10 \\ \hline
  \end{tabular}
  \caption[Final scoring table]{Table with ranks attributed to
    performance of each gene finder in several categories. The ranks
    for performance are 0 being not applicable, and 1 through 3 being increasing levels of performance. Since RefSeq is not publicly available, it is marked as N/A in the cumulative rank when considering availability. The dashed line separates categories associated with operation and use
    from categories describing gene prediction performance.}\label{table:final-score}
\end{table}


\section{Future Work}

While this work is extensive, there are still many areas that could be
expanded on. The sheer number of gene finding tools available means
that many more could be compared to the ones presented here. In
addition, supplying more training data to Braker2 may improve gene
finding performance, and training on RNAseq from different organisms
could be used to improve the performance of Braker2. With more data in mind, different types of training data --- such as RNAseq,
expressed sequence tags (ESTs), and protein sequences --- could also be
used as supporting evidence for training Braker2. Another
limitation of this work is that the gene finders were run with default
parameters, and it is possible that tuning the parameters of the gene
finders could improve performance. Exploration of newer gene finding models, such as those based on deep learning techniques, could also be
performed to see how they compare to the more established gene finding
tools.

Following the gene prediction process, there are a number of different
analyses that could be performed on the predicted genes. For example,
functional annotation of the predicted genes could be performed using
tools such as BLAST2GO to assign Gene Ontology (GO) terms and KEGG
pathways to the predicted genes. This could provide further insight
into the functional roles of the predicted genes and their potential
applications. More importantly in the context of this work, further analysis
of predicted genes with the antiSMASH tool could be used to
identify biosynthetic gene clusters (BGCs) from the predicted genes.

Expanding the number of genomes used in this work would also be
beneficial, as the results presented here are based on a limited
number of genomes. Including more genomes from different species and
strains of \textit{Trichoderma} could provide a more comprehensive
understanding of gene finding performance across the
genus. Additionally, comparing the performance of gene finders on
other fungal genomes could provide insights into whether the findings
are generalizable. Examining the performance of gene finders on
different assemblies of the same genome could also be useful, as
different assemblies may have different levels of completeness and
accuracy. This could help to identify tangible benefits of using one
assembly over another, and could also provide insights into the
limitations of gene finders when applied to different assemblies.

We also encourage the developers of gene finding tools to consider expanding the selection of genomes used in their testing and training datasets to include a more diverse set of fungal genomes. This could help to improve the performance of gene finders on fungal genomes, and may improve performance of gene finder in AT-rich regions, which are common in fungal genomes. In addition, we encourage developers to continue to improve the documentation and the usability of their tools, as ease of installation and use are important factors for many users. 

Finally, the results of this work could be used to inform future
research in the field of fungal genomics, particularly in the case of
the DC1 and Tsth20 genome assemblies. The results of these genome
assemblies are high-quality, and contain near-chromosomal scale
sequences. The predicted genes from these genomes are a rich resource
for future research, and could be used to identify novel genes and
gene families in DC1 and Tsth20. Further understanding of the
mechanisms behind salt and drought tolerance as well as degradation of
hydrocarbon in suboptimal environments could be achieved by studying
the predicted genes in these genomes. 
