\section{Region Identification}

The region identification process begins with identification of
regions sharing gene calls from each tool. For this project, we have
classified regions as complete, partial or singleton. A complete
region is a set of overlaps which contains a feature from each tool
considered in the region finding process. A partial region is a set of
overlaps which includes more than one but not all tools considered. A
singleton is a region in which a feature from only one tool is
present. Table \ref{table:regioncounts} displays the results from the
region finding process when applied to only the features of type
'gene' predicted by each tool.

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|c|c|c|}
      \hline
      Assembly & Regions (total) & Complete Agreement & Partial Agreement & Singletons\\ \hline
      DC1 & 11269 & 8483 & N/A & 2786  \\ \hline
      Tsth20 & 12272 & 8737 & N/A & 3535  \\ \hline
      \textit{T. reesei} & 9823 & 8282 & 557 & 984  \\ \hline
      \textit{T. harzianum} & 13388 & 8009 & 3314 & 2065  \\ \hline
      \textit{T. virens} & 12045 & 7537 & 3715 & 793  \\ \hline
    \end{tabular}
  \end{center}
  \caption{Counts of regions identified in total and total number of
    regions where a prediction from each individual tool was
    found. Partial agreement values for DC1 and Tsth20 are set as N/A
    as there were only two tools in consideration.}
  \label{table:regioncounts}
\end{table}

The results of the region finding process when applied to gene calls
show a mix of agreement and disagreement between the tools considered
here. While regions of complete agreement make up the majority of
regions in all assemblies, there are more partial agreements and
singletons than one would expect under the assumption that gene
finding tools are equal. Both DC1 and Tsth20 have a large number of
singleton regions present in comparison to the RefSeq datasets.


\section{Genes in Regions of Anomalous GC Content}

Evaluating gene finder performance in regions of anomalous GC content
is one of the key topics of this research. One simple way to evaluate
performance is whether or not gene finding tools predict genes
uniformly throughout a given sequence. Biologically, we know that
regions of anomalous nucleotide composition are less likely to contain
coding sequences than typical genomic regions, leading us to the
problem of first identifying predicted genes in standard and anomalous
regions. After identifying low GC segments within each assembly, we
can include them in the region identification method. From this
result, we classified predicted genes into two classes; genes in
regions with normal GC content, and genes in regions with anomalous
content. In this case, anomalous content is defined as a window of
genomic sequence containing a percent GC composition of 28\% or
lower. This number was chosen based on the plots of GC content
presented in the assembly section of the results. After classifying
predicted genes, two-sided binomial tests were performed with the null
hypothesis being that predicted genes are distributed uniformly
throughout an assembly. Framed differently, we expect the sum of genes
predicted in both regular and irregular regions to be proportional to
the sum of lengths of those regions, respectively. This is not the
case as demonstrated in table~\ref{table:gc-binomial}.

\begin{table}
  \begin{center}
    \begin{tabular}{|c|c|c|c|c|c|}
      \hline
      Tool & DC1 & Tsth20 & \textit{T. reesei} & \textit{T. harzianum} & \textit{T. virens} \\ \hline
      Braker2 & $9.56^-181$ & $1.14^-259$ & $2.68^-96$ & $4.05^-140$ & $1.35^-35$ \\ \hline
      GeneMark & $5.12^-216$ & $0.0$ & $5.66^-49$ & $5.37^-219$ & $5.31^-35$ \\ \hline
      RefSeq & N/A & N/A & $1.29^-49$ & $2.44^-205$ & $7.40^-33$ \\ \hline
    \end{tabular}
  \end{center}
  \caption{\textit{p} values produced from a two-sided binomial test
    for each combination of tool and assembly.}
  \label{table:gc-binomial}
\end{table}
