With an ever-increasing number of gene prediction tools available, and the continuous discovery of new and unique genomic sequences, it is important to assess the performance of these gene finding tools. This chapter will address the following research questions, which will be answered in the context of \textit{Trichoderma} genomes. The answers to these questions will provide insight into the performance of select gene finding tools, and may help inform users when selecting a gene prediction tool for use with fungal genomes and \textit{Trichoderma} genomes in particular.

\section{\textit{Trichoderma} Assembly Results}
\label{rq:assembly-results}
Since gene finding tools operate on an assembled genomic sequence, it
must follow that the results will be influenced by the supplied
assembly. Before applying gene finders to the new assemblies, we
should first evaluate the assemblies using general assembly metrics and compare with them existing assemblies. We ask: \textbf{how do assemblies of DC1 and
  Tsth20 compare to existing \textit{Trichoderma} assemblies?} With
these isolates being from the \textit{Trichoderma} family, we expect
assembly metrics to be similar in nature to existing assemblies from
NCBI, but are they? In addition, what are the quantitative features of these assemblies?

\section{Profiling of Gene Finding Tools}
\label{rq:gene-finding-tools}
Different gene finding tools may predict different types of features
associated with gene structures. The question arises: \textbf{which
  (if any) gene finders predict additional features outside the
  standard gene model?} Additional features in this case include
promoter sequences, transcription binding sites, activating sequences
and other upstream or downstream sequences. In addition, different
gene finding tools employ differing programming languages and
algorithms which raises several questions. \textbf{How are these gene
  finding tools implemented? Is the software straightforward to
  install? Are the tools user-friendly? What is the processing time
  and memory consumption of different gene finding tools in the
  context of \textit{Trichoderma}?}

\section{Number of Features Predicted}
\label{rq:number-of-features}  

One common method for evaluating gene finding tools is by comparing the number of genes and coding sequences each tool predicts. This gives insight into the sensitivity of each tool, as well as a sense of completeness of the predicted gene set. We will compare the number of features predicted by each tool, and ask: \textbf{do gene finders predict similar numbers of features in \textit{Trichoderma} genomes?}

\section{Lengths of Predicted Genes}
\label{rq:gene-lengths}
Genome assemblies can contain a wide range of gene lengths, and for some
users, genes of a specific length may be a key point of interest, as is the case in \textit{Trichoderma}, where genes encoding non-ribosomal peptide synthases are generally quite large~\cite{komaki2020}. Thus, it is important to assess the lengths of predicted genes from different gene finding tools. We will compare the lengths of predicted genes from each tool and ask: \textbf{do gene finders predict genes of similar lengths in \textit{Trichoderma}?}

\section{Agreement Between Gene Finders} 
\label{identify-regions}
With several sets of gene predictions, it is important to assess the agreement and disagreement between the predictions of different gene finding tools. To answer this question, we will identify `regions' of overlapping predictions. A 
region is defined as a set of overlapping gene predictions, where a gene prediction from one tool overlaps with a gene prediction from another tool.
With overlapping regions identified, we can determine any agreement or disagreement in predictions between gene finding tools
from which we can ask: \textbf{do gene finders agree on their
  predictions?} \textbf{If no, to what extent do they disagree?}

\section{Validation of Predicted Genes via InterProScan}
\label{rq:interproscan}
In an effort to validate, or at least provide supporting evidence for
any given gene prediction we will apply InterProScan to coding
sequences predicted by each of the gene finders to identify features
associated with protein function. Genes will be considered as `valid'
if the gene's protein sequence contains binding sites, motifs, or
other functional characteristics of proteins. Using the results from
InterProScan, we ask: \textbf{in the context of
  \textit{Trichoderma}, to what extent do proteins predicted by gene finders contain
  functional signatures? Do gene finders predict similar numbers of genes with functional signatures?}

\section{Similarity Searches of Predicted Genes with tblastn}
\label{rq:tblastn}
In a similar manner to validation via InterProScan, we can use protein alignments to search for similarity between the protein products of predicted genes and existing protein sequences. This will allow us to assess the degree of similarity between proteins from predicted genes and known proteins, which can provide further validation of gene predictions and help infer possible function. We ask: \textbf{do gene finders predict genes whose translations are similar to known proteins in \textit{Trichoderma}? Do gene finders identify comparable numbers of proteins that show similarity to known proteins?}

\section{Performance in AT-rich Genomic Sequence}
\label{rq:anomalous-sequence-content}
Results from the assembly process show that AT-rich (GC-poor) sequence content is prevalent in most \textit{Trichoderma} assemblies we selected. These regions of assemblies present an interesting opportunity to assess gene finding performance
in regions of anomalous GC content, since features such as transposable elements and repetitive sequences are often found in these regions. We ask: \textbf{do gene finders perform differently in AT-rich regions of \textit{Trichoderma} genomes? How well do gene finders perform in these regions?}   

\section{BUSCO Completeness}
\label{rq:busco-completeness}
The use of existing benchmarks for gene finding performance is useful
when assessing performance of gene finding tools, particularly in the
case of genes that should be evolutionarily conserved. \textbf{Do gene
  finders perform similarly when predicting conserved single-copy orthologs expected in fungal
  genomes?}

\section{Selection of a Gene Finding Tool}

With all the results generated, we can provide insight to the
question: \textbf{when studying \textit{Trichoderma}, which gene finding tool should one choose in any given context?}
