\section{Research Questions}

With an ever-increasing number of gene prediction tools available to
users, it is important to assess and understand their behaviour and
performance in the context, particularly in the context of new genome
assemblies of lesser studied organisms, where a reference prediction
set may not be available. The main purpose of this research is to
evaluate and compare gene finding tools in the context of
\textit{Trichoderma} assemblies where a gold standard set of gene
predictions does not exist. To assess behaviour and performance in
these contexts, we have defined X problems to profile the selected
gene finding tools. In addition to applying selected gene finding
tools to novel \textit{Trichoderma} isolates, Tsht20 and DC1, we also
applied selected gene finding tools to existing \textit{Trichoderma}
assemblies from the National Center for Biotechnology Information
(NCBI).

\section{\textit{Trichoderma} Assembly Results}
\label{rq:assembly-results}

Since gene finding tools operate on an assembled genomic sequence, it
must follow that the results will be influenced by the supplied
assembly. Before applying gene finders to the new assemblies, we
should first investigate the new assemblies by generating general
assembly metrics for the new assemblies to contrast and compare with
existing assemblies. We ask: \textbf{how do assemblies of DC1 and
  Tsth20 compare to existing \textit{Trichoderma} assemblies?} With
these isolates being from the \textit{Trichoderma} family, we expect
assembly metrics to be similar in nature to existing assemblies from
NCBI, but do they?

\section{Profiling of Gene Finding Tools}

Different gene finding tools may predict different types of features
associated with gene structures. The question arises: \textbf{which
  (if any) gene finders predict additional features outside of the
  standard gene model?} Additional features in this case include
promoter sequences, transcription binding sites, activating sequences
and other upstream or downstream sequences. In addition, different
gene finding tools employ differing programming languages and
algorithms which raises several questions. \textbf{How are these gene
  finding tools implemented? Is the software straightforward to
  install? Are the tools user-friendly? What is the processing time
  and memory consumption of different gene finding tools in the
  context of \textit{Trichoderma}?}

\section{Number of Features Predicted}

\textbf{Do gene finders predict similar numbers of features in the
  context of \textit{Trichoderma} genomes?} 
\label{rq:number-of-features}  
One common method for
evaluating gene finding tools is by looking at the number of features
predicted by each tool. These features make an obvious point of
comparison for selected gene finding tools. The term `feature' here is
somewhat ambiguous, referring to many possible categories of genomic
feature. For each gene finding tool, we compare the counts for
predicted genes, transcripts, and coding sequences.

\section{Lengths of Predicted Genes}
\label{rq:gene-lengths}

Genome assemblies can contain a wide range of gene lengths. For some
users, genes of a specific length may be a key point of interest, so
the ability of a gene prediction tool to capture the broad range of
possible gene lengths is another important metric for comparison. Thus
we ask the question: \textbf{do different gene finders predict genes
  of similar lengths in \textit{Trichoderma}?}

\section{Performance in Regions of Anomalous Sequence Content}
\label{rq:anomalous-sequence-content}

One of the inspirations for this research is the unique composition of
genomic sequence in \textit{Trichoderma}. Results from the assembly
process show that GC content in \textit{Trichoderma} strains is
abnormal throughout most assemblies. These regions of assemblies
present an interesting opportunity to assess gene finding performance
in regions of anomalous GC content. The question follows: \textbf{do
  gene finders behave differently in regions of anomalous sequence
  content?}

\section{BUSCO Completeness}

The use of existing benchmarks for gene finding performance is useful
when assessing performance of gene finding tools, particularly in the
case of genes that should be evolutionarily conserved. \textbf{Do gene
  finders predict conserved single-copy orthologs expected in fungal
  genomes?}

\section{Identifying Regions of Agreement and Disagreement} \label{identify-regions}

With predicted genes from several tools available, the question we
would like to ask is whether or not the gene finders agree with one
another for any given prediction. To answer this question, we will
identify 'regions' of overlapping predictions. A region can be defined
as a start and stop position of a set of individual or overlapping
features from one or more gene finding tools and external
sources. With regions identified, we can determine agreement, or more
importantly, disagreement in predictions between gene finding tools
from which we can ask: \textbf{do gene finders agree on their
  predictions?} \textbf{If no, to what extent do they disagree? Are
  there genomic regions where agreement or disagreement are more
  prevalent?}

\section{Validation of Predicted Genes via InterProScan}

In an effort to validate, or at least provide supporting evidence for
any given gene prediction we will apply InterProScan to coding
sequences predicted by each of the gene finders to identify features
associated with protein function. Genes will be considered as `valid'
if the gene's protein sequence contains binding sites, motifs, or
other fucntional characteristics of proteins. Using the results from
InterProScan, we ask the question: \textbf{in the context of
  \textit{Trichoderma}, do proteins predicted by gene finders contain
  functional signatures?}

%\section{Identification of a Core \textit{Trichoderma} Gene Set.}
%Using results from identification of regions in section
%\ref{identify-regions} we will identify a shared or `consensus' set of
%predictions for each \textit{Trichoderma} assembly. \textbf{Can we
%  identify a set of consensus predictions from gene finding results?
%  What are the properties of the resulting set?}

\section{Selection of a Gene Finding Tool}

With all the results generated, we can provide insight to the
question: \textbf{which gene finding tool should one choose?}
