\section{Research Questions}

With an ever-increasing number of gene prediction tools available to
users, it is important to assess and understand their behaviour and
performance in the context, particularly in the context of new genome
assemblies of lesser studied organisms, where a reference prediction
set may not be available. The main purpose of this research is to
evaluate and compare gene finding tools in the context of
\textit{Trichoderma} assemblies where a gold standard set of gene
predictions does not exist. To assess behaviour and performance in
these contexts, we have defined X problems to profile the selected
gene finding tools. In addition to applying selected gene finding
tools to novel \textit{Trichoderma} isolates, Tsht20 and DC1, we also
applied selected gene finding tools to existing \textit{Trichoderma}
assemblies from the National Center for Biotechnology Information
(NCBI).

\section{\textit{Trichoderma} Assembly Results}

\textbf{How do assemblies of DC1 and Tsth20 compare to existing
  \textit{Trichoderma} assemblies?} Since gene finding tools operate
on an assembled genomic sequence, it must follow that the results will
be influenced by the supplied assembly. With that said, before
applying gene finders to the new assemblies, we should first
investigate the new assemblies, by generating general assembly metrics
for the new assemblies to contrast and compare with existing
assemblies. These metrics include, total assembly length, number of
contigs, N50, L50, GC content and repeat content. With these
assemblies being individuals from \textit{Trichoderma} family, we
expect assembly metrics to be similar in nature to existing assemblies
from NCBI.

\section{Number of Features Predicted}

\textbf{Do gene finders predict similar numbers of features in the
  context of \textit{Trichoderma} genomes?} One common method for
evaluating gene finding tools is by looking at the number of features
predicted by each tool. The term 'feature' here is somewhat ambiguous,
referring to many possible categories of genomic feature. These
features make an obvious point of comparison for selected gene finding
tools. For each gene finding tool, we can compare the counts for
predicted genes, transcripts, and coding sequences.

\section{Lengths of Predicted Genes}

Genome assemblies can contain a wide range of gene lengths. For some
users, genes of a specific length may be a key point of interest, so
the ability of a gene prediction tool to capture the broad range of
possible gene lengths is another important metric for comparison. Thus
we ask the question; \textbf{Do gene different finders predict genes
  of similar lengths in \textit{Trichoderma}?}

\section{Performance in Regions of Anomalous Sequence Content}

One of the inspirations for this research is the unique composition of
genomic sequence in \textit{Trichoderma}. Results from the assembly
process show that GC content in \textit{Trichoderma} strains is
abnormal throughout most assemblies. These regions of assemblies
present an interesting opportunity to assess gene finding performance
in regions of anomalous GC content. The question follows; \textbf{do
  gene finders behave differently in regions of anomalous sequence
  content?}

\section{BUSCO Completeness}

\textbf{Do gene finders correctly predict conserved single-copy
  orthologs expected in fungal genomes?}


\section{Identifying Regions of Agreement and Disagreement}

With predicted genes from several tools available, the question we
would like to ask is whether or not the gene finders agree with one
another for any given prediction. To answer this question, we will
identify 'regions' of overlapping predictions. A region can be defined
as a start and stop position of a set of individual or overlapping
features from one or more gene finding tools and external
sources. With regions identified, we can determine agreement, or more
importantly, disagreement in predictions between gene finding tools
from which we can ask the question; \textbf{do gene finders agree on
  their predictions?}
